entity-framework/core/saving/transactions.md

```diff
@@ -14,19 +14,27 @@ Transactions allow several database operations to be processed in an atomic mann
 
 ## Default transaction behavior
 
-By default, if the database provider supports transactions, all changes in a single call to `SaveChanges()` are applied in a transaction. If any of the changes fail, then the transaction is rolled back and none of the changes are applied to the database. This means that `SaveChanges()` is guaranteed to either completely succeed, or leave the database unmodified if an error occurs.
+By default, if the database provider supports transactions, all changes in a single call to `SaveChanges` are applied in a transaction. If any of the changes fail, then the transaction is rolled back and none of the changes are applied to the database. This means that `SaveChanges` is guaranteed to either completely succeed, or leave the database unmodified if an error occurs.
 
 For most applications, this default behavior is sufficient. You should only manually control transactions if your application requirements deem it necessary.
 
 ## Controlling transactions
 
-You can use the `DbContext.Database` API to begin, commit, and rollback transactions. The following example shows two `SaveChanges()` operations and a LINQ query being executed in a single transaction.
+You can use the `DbContext.Database` API to begin, commit, and rollback transactions. The following example shows two `SaveChanges` operations and a LINQ query being executed in a single transaction:
 
-Not all database providers support transactions. Some providers may throw or no-op when transaction APIs are called.
+[!code-csharp[Main](../../../samples/core/Saving/Transactions/ControllingTransaction.cs?name=Transaction&highlight=2,16-18)]
 
-[!code-csharp[Main](../../../samples/core/Saving/Transactions/ControllingTransaction/Sample.cs?name=Transaction&highlight=3,17,18,19)]
+While all relational database providers support transactions, other providers types may throw or no-op when transaction APIs are called.
 
-## Cross-context transaction (relational databases only)
+## Savepoints
+
+When `SaveChanges` is invoked and a transaction is already in progress on the context, EF will automatically create a *savepoint* before saving any data. Savepoints are points within a database transaction which may be later be rolled back to, if an error occurs or for any other reason. If `SaveChanges` encounters any error, it will automatically roll the transaction back to the savepoint, leaving the transaction in the same state as if it had never started. This allows you to possibly correct issues and retry saving, in particular when [optimistic concurrency](xref:core/saving/concurrency) issues occur.
```

Comment:
```suggestion
When `SaveChanges` is invoked and a transaction is already in progress on the context, EF automatically creates a *savepoint* before saving any data. Savepoints are points within a database transaction which may be later be rolled back. For example, savepoints can roll back if an error occurs or for other reasons. If `SaveChanges` encounters an error:
* The transaction is automatically rolled back to the savepoint.
* The transaction stays in the same state as if it had never started.
Savepoints:
*  Allow the possibly to correct issues and retry saving.
* Are useful when [optimistic concurrency](xref:core/saving/concurrency) issues occur.
```

Reply:
Avoid future tense and long sentences.

Reply:
Definitely accept the future tense comment, need to get used to that.

But I looked at the text again in preview, and the last two sentences don't seem overly long to me... IMHO it's clearer (and shorter) as-is than broken down into bullet points...